solr/solr-ref-guide/src/nested-documents.adoc

```diff
@@ -0,0 +1,309 @@
+
+=== XML Examples
+
+For example, here are two documents and their child documents.
+It illustrates two styles of adding child documents; the first is associated via a field "comment" (preferred),
+and the second is done in the classic way now referred to as an "anonymous" or "unlabelled" child document.
+This field label relationship is available to the URP chain in Solr but is ultimately discarded.
+Solr 8 will save the relationship.
+
+[source,xml]
+----
+<add>
+  <doc>
+    <field name="id">1</field>
+    <field name="title">Solr adds block join support</field>
+    <field name="content_type">parentDocument</field>
+    <field name="content">
+      <doc>
+        <field name="id">2</field>
+        <field name="comments">SolrCloud supports it too!</field>
+      </doc>
+    </field>
+  </doc>
+  <doc>
+    <field name="id">3</field>
+    <field name="title">New Lucene and Solr release is out</field>
+    <field name="content_type">parentDocument</field>
+    <doc>
+      <field name="id">4</field>
+      <field name="comments">Lots of new features</field>
+    </doc>
+  </doc>
+</add>
+----
+
+In this example, we have indexed the parent documents with the field `content_type`, which has the value "parentDocument".
+We could have also used a boolean field, such as `isParent`, with a value of "true", or any other similar approach.
+
+=== JSON Examples
+
+This example is equivalent to the XML example above.
+Again, the field labelled relationship is preferred.
+The labelled relationship here is one child document but could have been wrapped in array brackets.
+For the anonymous relationship, note the special `\_childDocuments_` key whose contents must be an array of child documents.
+
+[source,json]
+----
+[
+  {
+    "id": "1",
+    "title": "Solr adds block join support",
+    "content_type": "parentDocument",
+    "comments": [{
+        "id": "2",
+        "content": "SolrCloud supports it too!"
+      },
+      {
+        "id": "3",
+        "content": "New filter syntax"
+      }
+    ]
+  },
+  {
+    "id": "4",
+    "title": "New Lucene and Solr release is out",
+    "content_type": "parentDocument",
+    "_childDocuments_": [
+      {
+        "id": "5",
+        "comments": "Lots of new features"
+      }
+    ]
+  }
+]
+----
+
+.Root Only Mode
+[NOTE]
+====
+ In Root-only schemas, these two documents will result in the same docs being indexed(Root-only schemas do not honor nested relationships).
+ When quried, child docs will be appended to _childDocuments_ key.
+====
+
+
+=== Querying Nested Documents
+
+ * `<<nested-documents.adoc#child-doc-transformer,[child]>>` Document Transformer
+ * <<nested-documents.adoc#children-query-parser,Block Join Children Query Parser>>
+ * <<nested-documents.adoc#parents-query-parser,Block Join Parent Query Parser>>
+ * <<nested-documents.adoc#combining-block-join-query-parsers-with-child-doc-transformer,Advanced Queries>>
+
```

Comment:
How about we at least mention that JSON Faceting also has special nested document support.  Just link to it; no need to do an example.  It's be nice if the mention included "blockChildren" and "blockParent" for searchability on this page.

Reply:
Why does this section start with links to other sections of the same and other pages without any context? Is the reader supposed to refer to those sections? Is this a secondary TOC (if so, we have another way to do that that requires less ongoing maintenance)?

Reply:
> (if so, we have another way to do that that requires less ongoing maintenance)?

How might that be done?

Reply:
You could add `{section-toc}`. It's described here: https://lucene.apache.org/solr/guide/7_6/how-to-contribute.html#table-of-contents. It will automatically create a mini-table of contents from the sub-headings of only this section. If sections are added in the future, then the list will automatically be updated during the conversion.

Also, just as a note, whenever you want to link to sections within the same page, you only need to put the section name in the double brackets (as in `<<Child Doc Transformer>>`) and don't need the fully qualified file name + section.

Reply:
I tried adding
```
== Searching Nested Documents
{section-toc}
```
, which did not seem to work.
Am I not doing this right?